sql/_34_fig/cbrd_24252/cases/cbrd_24252_22.sql

```diff
@@ -0,0 +1,68 @@
+set trace on;
+
+/* dummy data */
+drop table if exists dummy;
+create table dummy (col_a int);
+insert into dummy
+select rownum from
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+
+/* ----------------------------------------
+ *
+ * test case 22
+ *   - child c (parent_col_a:null, parent_col_b:null)
+ *   - parenta pa (c:parenta_col_a->pa:col_a)
+ *   - parentb pb (c:parentb_col_a->pb:col_a)
+ *
+ * ---------------------------------------- */
+
+drop table if exists t_child;
+drop table if exists t_parenta;
+drop table if exists t_parentb;
+create table t_parenta (col_a int primary key, col_b int);
+create table t_parentb (col_a int primary key, col_b int);
+create table t_child (col_a int auto_increment primary key, col_b int);
+alter table t_child add column parenta_col_a int references t_parenta (col_a); /* null */
+alter table t_child add column parentb_col_a int references t_parentb (col_a); /* null */
+insert into t_parenta select col_a, col_a from dummy;
+insert into t_parentb select col_a, col_a from dummy;
+insert into t_child select null, (col_a * -1), col_a, col_a from dummy;
+insert into t_child select null, (col_a * -1), col_a, col_a from dummy;
+insert into t_child select null, (col_a * -1), null, null from dummy limit 1; /* col_a:2001 */
+
+/* ansi‑style */
+select /*+ recompile */
+    c.col_a,
+    c.col_b
+from
+    t_child as c
+    inner join t_parenta as pa on c.parenta_col_a = pa.col_a
+    inner join t_parentb as pb on c.parentb_col_a = pb.col_a
+where
+    c.col_b = -1;
+show trace;
```

Comment:
@youngjinj 
The trace result executed with csql is different from the attached scenario on CBRD-24252.

Reply:
The attached scenario on CBRD-24252 is incorrect.
The answer in this PR is correct.

Reply:
Okay thanks for the explanation.